packages/components/src/query-controls/terms.ts

```diff
@@ -3,15 +3,20 @@
  */
 import { groupBy } from 'lodash';
 
+/**
+ * Internal dependencies
+ */
+import type { Entity, TermsWithChildren } from './types';
+
 /**
  * Returns terms in a tree form.
  *
- * @param {Array} flatTerms Array of terms in flat format.
+ * @param flatTerms Array of terms in flat format.
  *
- * @return {Array} Array of terms in tree format.
+ * @return Terms in tree format.
  */
-export function buildTermsTree( flatTerms ) {
-	const flatTermsWithParentAndChildren = flatTerms.map( ( term ) => {
+export function buildTermsTree( flatTerms?: readonly Entity[] ) {
```

Comment:
Without `readonly`, there are 4 TS errors in `query-controls/test/terms.ts`:

<img width="1171" alt="Screen Shot 2022-12-21 at 5 50 23 PM" src="https://user-images.githubusercontent.com/4063887/209025089-53f58d68-1e62-4695-a423-9f76d153357c.png">